### ISSUE TITLE
Run WIndows tests on Appveyor

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8593]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#8593|
|Type|enhancement|
|Created|2016-07-12 15:05:37Z|
|Branch|https://github.com/twisted/twisted/tree/8593-appveyor|

as this is the only public CI with Windows support and we should allow contributors to get a quick feedback regarding the tests and coverage on Windows

it this fails, we might need to allow our buildbot to run tests based on PR from forks

<details><summary>Searchable metadata</summary>

```
trac-id__8593 8593
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__8593_appveyor 8593-appveyor
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1468335937264379 1468335937264379
changetime__1482286262034183 1482286262034183
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

https://github.com/twisted/twisted/pull/320

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban removed owner|
|-|-|

I set up an initial Appveyor build.

It used the python version provided by Appveyor as I tried to keep things as simple as possible.

Some tests are failing, but I think that they should be fixed in a separate ticket.

The Appveyor is configured to run the test but to ignore errors... so appveyor is still an unsupported platform.

The main reason why I added Appveyor is to allow PR from non-team members to have the tests executed on Windows and coverage reported for Windows.

Please check that changes are ok.

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_3"></a>|@markrwilliams commented|
|-|-|

The configuration changes are reasonable, and most importantly the Appveyor build is passing!

The buildbot failures are apparently unrelated to this PR (e.g. twistedchecker failures in unrelated files and various build bots that fail here and in other PRs as well.)

This looks ready for merging to me.

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_4"></a>|@markrwilliams commented|
|-|-|

I'm putting this back into review because I signed off on this before the tests finished.  Adi is still working on this.

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_5"></a>|@markrwilliams commented|
|-|-|

The configuration still looks reasonable, but now the builds are passing :)

Should I open a ticket for Appveyor and Python 3 given https://github.com/twisted/twisted/pull/320/files/c0955f13f2f2389ca8f27c32c3fbdedb53c57d2c..d3f20cb25fd432323c92613a9f81ac10424cd708 ?

Supposing all the builds pass after a merge from trunk, I'll say this should be merged.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban set owner to @adiroiban<br>@adiroiban set status to `assigned`|
|-|-|

we first need a tox env for python3 + windows ... but maybe this can be done in the same branch in which we enable py3 on appveyor... so yes. we need at least one ticket ... [#8611](https://github.com/twisted/twisted/issues/8611)

We also need a follow up ticket to fix the tests which pass on our buildslave but are failing on appveyor [#8612](https://github.com/twisted/twisted/issues/8612)

But I think that we should first have [#8214](https://github.com/twisted/twisted/issues/8214) solved.

I have updated the branch to reference the follow up tickets.

Thanks for the review.

Will merge.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|GitHub set status to `closed`|
|-|-|

In changeset 62e2b2019da3e20d82777eaf15aff05c503c8cb3
```
#!CommitTicketReference repository="" revision="62e2b2019da3e20d82777eaf15aff05c503c8cb3"
Merge 8593-appveyor: Run Python 2.7 Windows tests on Appveyor.


Author: adiroiban
Reviewer: markrwilliams
Fixes: #8593
```

**Comment by twisted-trac** [comment]
|<img alt="Tom Prince's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|Tom Prince commented|
|-|-|

In changeset 71643ca93e024d33dba8de9eef149876554c2dd7
```
#!CommitTicketReference repository="" revision="71643ca93e024d33dba8de9eef149876554c2dd7"
Merge pull request #636 from twisted/8953-old-zope-interface

Author: tomprince
Reviewer: rodrigc
Fixes: ticket:8593

Remove some compatability code for pre-4.0 versions of zope.interface.
```
